jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java

```diff
@@ -453,13 +461,10 @@ public Deflater getDeflater(Request request, long content_length)
             LOG.debug("{} excluded not gzip accept {}",this,request);
             return null;
         }
-        
-        Deflater df = _deflater.get();
-        if (df==null)
-            df=new Deflater(_compressionLevel,true);        
-        else
-            _deflater.set(null);
-        
+
+        Deflater df = _deflaterPool.acquire();
+        df.setLevel(_compressionLevel);
```

Comment:
So definitely do not set `compressionLevel` after acquire.
Instead, add a `if (isStarted()) throw new IllegalStateException(state())` to the setter for `compressionLevel`